utils/s2n_array.c

```diff
@@ -88,6 +88,40 @@ void *s2n_array_get(struct s2n_array *array, uint32_t index)
     return element;
 }
 
+void *s2n_array_insert(struct s2n_array *array, uint32_t index)
+{
+    if (array == NULL) {
+        return NULL;
+    }
+
+    if (array->num_of_elements >= array->capacity) {
+        /* Enlarge the array */
+        GUARD_PTR(s2n_array_enlarge(array, array->capacity * 2));
+    }
+
+    memcpy_check_ptr((uint8_t *) array->elements + array->element_size * (index + 1),
+                     (uint8_t *) array->elements + array->element_size * index,
+                     (array->num_of_elements - index) * array->element_size);
+
+    void *element = (uint8_t *) array->elements + array->element_size * index;
+    array->num_of_elements++;
+
+    return element;
+}
+
+int s2n_array_delete(struct s2n_array *array, uint32_t index)
+{
+    notnull_check(array);
+
+    memcpy_check((uint8_t *) array->elements + array->element_size * index,
```

Comment:
after we move this data down, should we overwrite what was previously at the end with zeros?

Reply:
Done.